bitonic_sort.h

```diff
@@ -0,0 +1,6174 @@
+static __inline void BITONIC_SORT_28(SORT_TYPE *dst) {
+  SORT_CSWAP(dst[1], dst[2]);
+  SORT_CSWAP(dst[3], dst[4]);
+  SORT_CSWAP(dst[5], dst[6]);
+  SORT_CSWAP(dst[8], dst[9]);
+  SORT_CSWAP(dst[10], dst[11]);
+  SORT_CSWAP(dst[12], dst[13]);
+  SORT_CSWAP(dst[15], dst[16]);
+  SORT_CSWAP(dst[17], dst[18]);
+  SORT_CSWAP(dst[19], dst[20]);
+  SORT_CSWAP(dst[22], dst[23]);
+  SORT_CSWAP(dst[24], dst[25]);
+  SORT_CSWAP(dst[26], dst[27]);
+  SORT_CSWAP(dst[0], dst[2]);
+  SORT_CSWAP(dst[3], dst[5]);
+  SORT_CSWAP(dst[4], dst[6]);
+  SORT_CSWAP(dst[7], dst[9]);
+  SORT_CSWAP(dst[10], dst[12]);
+  SORT_CSWAP(dst[11], dst[13]);
+  SORT_CSWAP(dst[14], dst[16]);
+  SORT_CSWAP(dst[17], dst[19]);
+  SORT_CSWAP(dst[18], dst[20]);
+  SORT_CSWAP(dst[21], dst[23]);
+  SORT_CSWAP(dst[24], dst[26]);
+  SORT_CSWAP(dst[25], dst[27]);
+  SORT_CSWAP(dst[0], dst[1]);
+  SORT_CSWAP(dst[4], dst[5]);
+  SORT_CSWAP(dst[2], dst[6]);
+  SORT_CSWAP(dst[7], dst[8]);
+  SORT_CSWAP(dst[11], dst[12]);
+  SORT_CSWAP(dst[9], dst[13]);
+  SORT_CSWAP(dst[14], dst[15]);
+  SORT_CSWAP(dst[18], dst[19]);
+  SORT_CSWAP(dst[16], dst[20]);
+  SORT_CSWAP(dst[21], dst[22]);
+  SORT_CSWAP(dst[25], dst[26]);
+  SORT_CSWAP(dst[23], dst[27]);
+  SORT_CSWAP(dst[0], dst[4]);
+  SORT_CSWAP(dst[1], dst[5]);
+  SORT_CSWAP(dst[7], dst[11]);
+  SORT_CSWAP(dst[8], dst[12]);
+  SORT_CSWAP(dst[6], dst[13]);
+  SORT_CSWAP(dst[14], dst[18]);
+  SORT_CSWAP(dst[15], dst[19]);
+  SORT_CSWAP(dst[21], dst[25]);
+  SORT_CSWAP(dst[22], dst[26]);
+  SORT_CSWAP(dst[20], dst[27]);
+  SORT_CSWAP(dst[0], dst[3]);
+  SORT_CSWAP(dst[2], dst[5]);
+  SORT_CSWAP(dst[7], dst[10]);
+  SORT_CSWAP(dst[9], dst[12]);
+  SORT_CSWAP(dst[14], dst[17]);
+  SORT_CSWAP(dst[16], dst[19]);
+  SORT_CSWAP(dst[21], dst[24]);
+  SORT_CSWAP(dst[23], dst[26]);
+  SORT_CSWAP(dst[13], dst[27]);
+  SORT_CSWAP(dst[1], dst[3]);
+  SORT_CSWAP(dst[2], dst[4]);
+  SORT_CSWAP(dst[8], dst[10]);
+  SORT_CSWAP(dst[9], dst[11]);
+  SORT_CSWAP(dst[0], dst[7]);
+  SORT_CSWAP(dst[5], dst[12]);
+  SORT_CSWAP(dst[15], dst[17]);
+  SORT_CSWAP(dst[16], dst[18]);
+  SORT_CSWAP(dst[22], dst[24]);
+  SORT_CSWAP(dst[23], dst[25]);
+  SORT_CSWAP(dst[14], dst[21]);
+  SORT_CSWAP(dst[19], dst[26]);
+  SORT_CSWAP(dst[2], dst[3]);
+  SORT_CSWAP(dst[9], dst[10]);
+  SORT_CSWAP(dst[1], dst[8]);
+  SORT_CSWAP(dst[4], dst[11]);
+  SORT_CSWAP(dst[6], dst[12]);
+  SORT_CSWAP(dst[16], dst[17]);
+  SORT_CSWAP(dst[23], dst[24]);
+  SORT_CSWAP(dst[15], dst[22]);
+  SORT_CSWAP(dst[18], dst[25]);
+  SORT_CSWAP(dst[20], dst[26]);
+  SORT_CSWAP(dst[0], dst[14]);
+  SORT_CSWAP(dst[2], dst[9]);
+  SORT_CSWAP(dst[1], dst[7]);
+  SORT_CSWAP(dst[3], dst[10]);
+  SORT_CSWAP(dst[6], dst[11]);
+  SORT_CSWAP(dst[16], dst[23]);
+  SORT_CSWAP(dst[15], dst[21]);
+  SORT_CSWAP(dst[17], dst[24]);
+  SORT_CSWAP(dst[20], dst[25]);
+  SORT_CSWAP(dst[12], dst[26]);
+  SORT_CSWAP(dst[2], dst[8]);
+  SORT_CSWAP(dst[4], dst[10]);
+  SORT_CSWAP(dst[16], dst[22]);
+  SORT_CSWAP
```

Comment:
This needs to be `1LL` or `1ULL`, otherwise it will overflow if `(63 - CLZ(size - 1)) > 32`.

Reply:
OK.